File: OnlineExamSystemFrontend/src/app/admin.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { User } from './user.service';
import { Question } from './question.service';


@Injectable({
  providedIn: 'root'
})
export class AdminService {


  deleteUser(username: string) {
    return this.httpClient.delete<void>("http://localhost:8080/admin/deleteUser/"+username);
  }


  createExam(questions: Question[]) {
    return this.httpClient.post<void>("http://localhost:8080/admin/createExam",questions);
  }


  
  constructor(private httpClient: HttpClient) { }

  getAllUsers() {
    return this.httpClient.get<User[]>("http://localhost:8080/admin/getAllUsers");
  }

  
}

File: OnlineExamSystemBackend/src/main/java/com/jbk/oes/services/AdminService.java
package com.jbk.oes.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.jbk.oes.dao.QuestionDAO;
import com.jbk.oes.dao.UserDAO;
import com.jbk.oes.modal.Question;
import com.jbk.oes.modal.User;

@Service
public class AdminService {

	@Autowired
	private UserDAO userDAO;
	
	@Autowired
	private QuestionDAO questionDAO;
	
	
	public List<User> getAllUsers() {
		return userDAO.getAllUsers();
	}


	public Boolean createExam(List<Question> questions) {
		
		Boolean status = questionDAO.createExam(questions);
		
		return status;
	}


	public void deleteUser(String username) {
		userDAO.deleteUser(username);		
	}
	
	 
}
